include/flash_map.h

```diff
@@ -50,34 +55,98 @@ struct flash_sector {
 	size_t fs_size;
 };
 
-/*
+/**
  * Start using flash area.
+ *
+ * @param id Flash area id
+ * @param fa Flash area.
+ *
+ * @return 1 On success. -ENODEV if no driver match.
  */
 int flash_area_open(u8_t id, const struct flash_area **fa);
 
+/**
+ * Close a flash area
+ *
+ * @param fa Flash area
+ */
 void flash_area_close(const struct flash_area *fa);
 
-/*
+/**
  * Read/write/erase. Offset is relative from beginning of flash area.
+ *
+ * @param fa Flash area.
+ * @param off Offset
+ * @param dst Destination
+ * @param len Length
+ *
+ * @return 1 On success. -ENODEV if no driver match.
  */
 int flash_area_read(const struct flash_area *fa, off_t off, void *dst,
 		    size_t len);
+
+/**
+ * Write to a flash area
+ *
+ * @param fa Flash area
+ * @param off Offset
+ * @param src Source to be read from
+ * @param len Data length
+ *
+ * @return 1 On success
+ */
 int flash_area_write(const struct flash_area *fa, off_t off, const void *src,
 		     size_t len);
+
+/**
+ * Erase a flash area
+ *
+ * @param fa Flash area
+ * @param off Offset
+ * @param len Lendth of data to be erased
+ *
+ * @return 1 On success
+ */
 int flash_area_erase(const struct flash_area *fa, off_t off, size_t len);
-/* int flash_area_is_empty(const struct flash_area *, bool *); */
 
-/*
+/**
  * Alignment restriction for flash writes.
+ *
+ * @param fa Flash area.
+ * @return Block size
  */
 u8_t flash_area_align(const struct flash_area *fa);
 
-/*
+/**
  * Given flash area ID, return info about sectors within the area.
+ *
+ * @param fa_id Flash area ID
+ * @param count Sector count
```

Comment:
Capacity of sectors array on input. Exists sectors count on successful output.

Reply:
second sentence is stil valid.

Reply:
I am not sure exactly what you mean.

Reply:
I omitted your response. I mean that count is [in, out] parameter:
```suggestion
 * @param[in, out] count Capacity of `sectors` array on input. Count of fetched sectors on output.
 * @param[out] sectors array
```